public/app/core/components/Select/UserPicker.test.tsx

```diff
@@ -1,14 +1,17 @@
 import React from 'react';
-import { render, screen } from '@testing-library/react';
+import { act, render, screen } from '@testing-library/react';
 import { UserPicker } from './UserPicker';
 
 jest.mock('@grafana/runtime', () => ({
   getBackendSrv: () => ({ get: jest.fn().mockResolvedValue([]) }),
 }));
 
 describe('UserPicker', () => {
-  it('renders correctly', () => {
-    render(<UserPicker onSelected={() => {}} />);
+  it('renders correctly', async () => {
+    await act(async () => {
+      render(<UserPicker onSelected={() => {}} />);
+    });
+
     expect(screen.getByTestId('userPicker')).toBeInTheDocument();
```

Comment:
```suggestion
  it('renders correctly', async () => {
    render(<UserPicker onSelected={() => {}} />);
    expect(await screen.findByTestId('userPicker')).toBeInTheDocument();
```

I think it could be fixed in a simpler way (at least I don't get the warning locally anymore). Using await/act/async combination is usually considered an antipattern.

Reply:
Yeah, I'm not completely sure either way to be honest. 

"Semantically", I'm not sure if `await screen.findByTestId('userPicker')` - we're not waiting for that selector to be present as it's already present on the initial render. It only seems to suppress the warning as an implementation side effect?

But then also I'm not sure if it makes sense that how I'm using `act()` suppresses this warning here also?

To be honest - RTL/act/useEffect is the blind spot so I'm very much eager to know how to do this properly 😅

Reply:
My reasoning is that when we use `await find*` we query for a content after all the async operations inside the component have ended. Here's another approach to fixing this: https://kentcdodds.com/blog/fix-the-not-wrapped-in-act-warning#an-alternative-waiting-for-the-mocked-promise - I think we actually use it in our code somewhere but can't remember where 🤔